### DISCUSSION TITLE
Correct query.. or parse return.. please help.

### DISCUSSION THREAD

**Discussion by zimart** [Budibase/budibase#14451] [Help] [answered]
Hello and welcome.
Sorry for what may be a simple question, but how to properly parse a sql query.
I'm new to Budibase.
I am using Mysql.
My sample query:

SELECT cenakatalogowa FROM c_sc where indeks='SCR-FME90-ZIIIP-95-SERGE-1250-1500';

More specifically, I would like to get the price of a product with a given SKU from the products table.
SELECT cenakatalogowa FROM c_sc where indeks={{skugen}}; 

I have an answer for example : 
{
  "cenakatalogowa": "1379.24"
}

How to properly construct a query - or parse the result using js - to obtain the expected result - e.g. the price itself in the input or in the table.

---> 1379,24

Thank you in advance.

**Comment by mikesealey** [comment, marked as answer]
Hey @zimart 

Is the puzzle that you're getting a string back from your query it should be a number? 

If so, in the Data tab when configuring your query, click "Schema" in query results, and you'll have some options to change the data type. (This image is taken from an old project where I followed along to a tutorial, so query names and exact structures might not match your specific use case)
![image](https://github.com/user-attachments/assets/7f33fbcd-d6de-4478-ad2c-bc8a2e5077e0)

If you want to take that value returned and use it in a form field, you could set it in state:

In this example, it's set with a button click. Add an action to the button click to run the query
![image](https://github.com/user-attachments/assets/1770d3c4-81b4-4b84-b31c-52c88e4d5e65)

Then add another on-click action to Update State
![image](https://github.com/user-attachments/assets/cd7417a5-87cf-497f-9fb6-8c015e752832)

In the bindings drawer, you'll now see that you have access to the Actions binding
![image](https://github.com/user-attachments/assets/a28078c2-32e9-4d85-80bf-446daf83b955)

And you can set the value of state to be the price you got back from the query.
![image](https://github.com/user-attachments/assets/71d63a8a-f84a-4857-981c-d3ab5c2329a3)

And then you can use that state variable to populate a form input (or anywhere else you can use the State binding :D)
![image](https://github.com/user-attachments/assets/ea28c655-d5c0-4738-af10-2e7f1e3373bb)


Hope this helps. If you've got any other questions, or if I've just completely misunderstood, please reach out and I'll see what I can do.

**Reply by midland192** [reply]
Hi

I'm following this example as I'm trying to do the same.
However on the last step where I'm trying to get the value from the state.variable I'm getting the JSON text

e..g.
{"data":[{"salesman_id":13}]}  

All I want to return is the value 13 so I can use this in my form

Am I missing something
